packages/react-scripts/config/env.js

```diff
@@ -29,7 +29,12 @@ function getClientEnvironment(publicUrl) {
       // For example, <img src={process.env.PUBLIC_URL + '/img/logo.png'} />.
       // This should only be used as an escape hatch. Normally you would put
       // images into the `src` and `import` them in code to get their paths.
-      'PUBLIC_URL': publicUrl
+      'PUBLIC_URL': publicUrl,
+      // Useful for allowing the hot dev websocket to connect to the host
+      // specified in the env file, instead of window.location, which could
+      // be wrong if the dev server is being proxied.
+      'HOST': process.env.HOST || '',
+      'PORT': process.env.PORT || ''
```

Comment:
Hi, can we make this not introduce additional environment variables?
Can we inject this code into the bundle elsehow?
We can use some black-boxed logic/variables since they'll be references from `react-dev-utils` and not the application itself.